File: 2020/day1.rb
input = ''

if __FILE__ == $0
  p input.split("\n").map(&:to_i).combination(2) {|a| p(a.inject(:*)) if a.inject(:+) == 2020}
  p input.split("\n").map(&:to_i).combination(3) {|a| p(a.inject(:*)) if a.inject(:+) == 2020}
end

File: 2018/day1.js
export const solve = (input) => {
    let numbers = input.split(/\n/)
    let results = new Set(); 
    var sum = 0;
    var i = 0
    while(true)
    {
        sum += parseInt(numbers[i]);   
        if(results.has(sum) === true ){
            return sum;
        }
        results.add(sum);
        console.log(sum);
        i++
        if(i === numbers.length)
            i=0;
    }
};

